--- 
title: "Analysing CRISPR Screens with edgeR"
author: "Göknur Giner"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography:
- book.bib
- packages.bib
description: |
  This is a book version to write a book.
  set in the _output.yml file.
  The HTML output format for this example is bookdown::gitbook,
link-citations: yes
github-repo: "rstudio/bookdown-demo"
---

# Welcome {-}
Welcome to the home of "Analyzing CRISPR Screens with edgeR". This platform serves as the central hub for a comprehensive guide on leveraging one of the most commonly used differential expression analysis Bioconductor package edgeR, for the analysis of CRISPR screens. Whether you're delving into CRISPR experiments for the first time or seeking advanced insights, this guide will equip you with essential skills and knowledge to extract meaningful information from your data.

## What You Will Learn{-}
Our mission is to provide you with a robust foundation in the analysis of CRISPR data using edgeR. Throughout this guide, we walk you through various steps of typical workflows using example data sets to check the quality of sequencing data, count the single guide RNAs (sgRNAs), process, analyze, visualize, and explore the differential abundance of the sgRNAs and related the pathways. By doing so, we address key concepts and methodologies, ensuring a deep understanding of the intricacies involved in CRISPR data analysis.

## What You Won’t Learn{-}
The landscape of CRISPR data analysis is vast and diverse. In this guide, we focus on specific tool—edgeR provided by Bioconductor. While we aim to cover edgeR comprehensively, it's important to note that the realm of bioinformatic analysis offers numerous possibilities, depending on the experimental setup and biological system. Our guide provides a starting point for your exploration, acknowledging the richness of available tools beyond the scope of this book.

Recognize that your specific data may require a tailored approach, and our examples serve as templates rather than one-size-fits-all solutions. We encourage you to view them as a foundation upon which you can build and adapt for your unique analyses.

## Who We Wrote This For{-}

This guide is crafted with both novice and experienced researchers in mind. Whether you're an experimental biologist stepping into the realm of CRISPR data or a seasoned bioinformatician seeking insights into edgeR in the context of CRISPR gene editing, you'll find valuable content here. We make minimal assumptions about your previous programming or statistical experience, aiming to create a resource that is accessible to a broad audience.

We welcome feedback from all users to improve this guide continually, enhancing accessibility and refining technical details. Your input is instrumental in making this resource more valuable to the community.

## Why We Wrote This{-}

The inception of this guide dates back to the recognition of the growing prominence of CRISPR data analysis within the field of biology. As CRISPR experiments evolved, so did the need for specialized tools. In response, we embarked on creating a guide that not only showcases the capabilities of edgeR but also celebrates the Bioconductor community's contributions to advancing the field of biological analysis. Our aim is to empower researchers like you with the tools and knowledge needed to navigate the complex landscape of CRISPR data analysis.

# Setting up{-}

## Install R{-}

R can be installed from the [R-project website](http://www.r-project.org/).

Direct downloads for all platforms can be downloaded at the links below:

[Mac](https://cran.r-project.org/bin/macosx/)

[Windows](https://cran.r-project.org/bin/windows/base/)

[Linux](https://cran.r-project.org/bin/linux/ubuntu/fullREADME.html)

## Install RStudio{-}

RStudio can be installed from the [RStudio website](http://www.rstudio.com/).

RStudio can be downloaded for all platforms at the link below:

<https://rstudio.com/products/rstudio/download/>

## Install required packages{-}
```{r, setup, include=FALSE, echo=FALSE}
knitr::opts_chunk$set(comment = '', 
                      fig.width = 6, 
                      fig.height = 6, 
                      message=FALSE,
                      warning=FALSE,
                      tidy = TRUE
)
```
```{r install packages, eval=FALSE, echo=-1}
setwd("~/Library/CloudStorage/GoogleDrive-goknurginer@gmail.com/My Drive/Professional/Workshops_given_by_me/BioC/Analysing-CRISPR-screens-with-edgeR")
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("edgeR")
BiocManager::install("GenomicAlignments")
BiocManager::install("Rsubread")
BiocManager::install("Biostrings")
BiocManager::install("AnnotationDbi")
BiocManager::install("org.Hs.eg.db")
install.packages("ggplot2")
install.packages("DT")
install.packages("ggrepel")
```
## Load required packages{-}
```{r load packages}
library(edgeR)
library(GenomicAlignments)
library(Rsubread)
library(Biostrings)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(ggplot2)
library(DT)
library(ggrepel) # tidy volcano
```
## Download the material used in this book {-}
1. Fastq files
2. GECKO human library

# Counting the sgRNAs with RSubread{-}
Then we load the guide library, and subsequently, we must convert it into the FASTA format to construct an index for alignment.
```{r read sgRNA library}
GeCKO <- read.delim("sgRNA_library/GeCKOv21_Human.tsv")
GeCKO[1:2,]
sgRNAs <- DNAStringSet(GeCKO$seq)
names(sgRNAs) <- GeCKO$UID
sgRNAs
```

```{r, eval=FALSE, echo=FALSE}
dir.create("./index")
writeXStringSet(sgRNAs,
                file="./index/GeCKO.fa")
```

## Building an index{-}
```{r build index, eval=FALSE}
buildindex("./index/GeCKO","./index/GeCKO.fa", 
           indexSplit=FALSE)
```
## Aligning the fastq files{-}
```{r, eval=FALSE}
fastqs <- dir(path = "./fastq_files", pattern = "*.fastq.gz", full.names = TRUE)
counts <- list()
mapping_results <- list()
for(i in 1:length(fastqs)){
  mapping_results[[i]] <-align("./index/GeCKO",fastqs[i],output_file = gsub(".fastq.gz",".bam",fastqs[i]),
                  nthreads=4,unique=TRUE,nBestLocations=1,type = "DNA",TH1 = 1,
                  maxMismatches = 0,indels = 0)
  temp <- readGAlignments(gsub(".fastq.gz",".bam",fastqs[i]))
  counts[[i]] <- data.frame(table(seqnames(temp[width(temp) == "20"])),row.names = "Var1")
}
my_counts <- do.call(cbind,counts)
colnames(my_counts) <- c("Control_1","Control_2","ToxA_1","ToxA_2","ToxB_1","ToxB_2")
```
## Creating a DGEList object{-}
```{r create dgelist object, eval=-8}
counts <- read.table("my_counts.txt", header = TRUE)
group <- factor(c("Control","Control","ToxA","ToxA","ToxB","ToxB"),levels = c("Control","ToxA","ToxB"))
samples <- data.frame(group = group, sampleName=colnames(counts), biorep=rep(c(1,2),3))
genes <- GeCKO
names(genes)[names(genes)=="gene_id"] <- "Symbol"
d <- DGEList(counts = counts, samples = samples, genes = genes)
d
dir.create("RData")
save(d, file="./RData/DGEList.RData")
```

# Preprocessing the data{-}
```{r load dgelist object}
load("./RData/DGEList.RData")
```

## Guide distribution across samples {-}
```{r guide distribution}
data <- NULL
sample_names <- colnames(d)
l <- colSums(!d$counts==0)
data <- data.frame(row.name = sample_names, l=l,t=as.character(d$samples$group))
data <- data[order(data$l, d$samples$group),]
col <- c("#6DAEB0", "#FAA578", "#FFB6C1")
my_bar <- barplot(data$l, 
                  xlim = c(0,70000),
                  names.arg = data$row.name, 
                  las=1, 
                  space=1, 
                  xlab = "The number of detected guides",
                  main = "The distribution of guides",
                  col = col[as.factor(data$t)], 
                  border=NA,
                  legend=TRUE,
                  horiz = TRUE)
text(my_bar, x=data$l+3000 # adjust this number
     , paste0("(n:", data$l, ")"),las = 2)
legend("bottomright",legend = levels(factor(data$t)),fill = col)
```

## Gini indices{-}
```{r gini}
boxplot(gini(d$counts) ~ d$samples$sampleName, ylab = "Gini Indices", xlab="", cex.axis=0.8)
```

## Library sizes by group {-}
```{r library sizes before filtering}
yy <- d[, order(d$samples$group, colSums(d$counts))]
yy.logcpm <- cpm(yy$counts, log = T)
boxplot(yy.logcpm, las = 2, pch = 19, col = col[as.factor(yy$samples$group)], outcex = 0.3, outcol = col[as.factor(yy$samples$group)],labels = colnames(yy.logcpm), main = "SgRNA abundance before filtering", ylab = "SgRNA representation (log2 cpm normalised reads)")
legend("topleft", legend = levels(factor(yy$samples$group)), col = col, pch = 19, cex = 0.5)
```

## Removing lowly expressed guides{-}
Guides lacking a substantial number of reads in any sample should be excluded from downstream analyses. There are multiple rationales for this choice. From a biological perspective, guides that fail to register at a biologically relevant level in any condition are deemed uninteresting and are consequently disregarded. From a statistical standpoint, the elimination of guides with low counts enhances the reliability of estimating the mean-variance relationship in the data. It also reduces the number of statistical tests required for subsequent analyses focused on differential abundance.

To implement this, here we used two filtering methods: one permissive and the other strict. For the rigorous approach, guides with low expression were filtered out, retaining as many guides as possible with meaningful counts. This was achieved using the filterByExpr() function in the edgeR package, a tool commonly applied in RNA-seq experiments for automated gene filtering. Initially, the count matrix contained 122,411 guides. After applying filterByExpr(), 113,395 guides (93%) were removed due to their low presence, leaving 9,016 guides for further analysis. It's worth noting that filterByExpr() often retains only one guide per gene, which is suboptimal for CRISPR screens, where the collective behavior of multiple guides targeting the same gene is essential for validation.

To address this limitation, we used a more permissive filtering approach. This method involved defining guide-wise threshold (th1) and sample-wise threshold (th2). Th1 was computed using count per million (CPM) normalized values, while th2 was determined as the average number of samples in each condition. This approach aims to strike a balance, allowing for the inclusion of as many guides as possible targeting the same gene, facilitating a more robust assessment of the collective impact of guides in CRISPR screens. Consequently, with the permissive approach, we retained 58171 (48%) of the guides.

### Remove Non-targeting guides
First we remove 1000 Non-targeting sgRNAs.
```{r non-targeting filtering}
d.raw <- d
dim(d <- d[!d$genes$Symbol %in% grep("Non", d$genes$Symbol, value = TRUE),])
```
### Strict filtering{-}
```{r strict filtering}
keep.exprs <- filterByExpr(d, group = d$samples$group)
table(keep.exprs)
dim(d.filtered.s <- d[keep.exprs, , keep.lib.sizes = FALSE])
```
```{r strict filtering guide distribution}
genes_sgrna <- d.filtered.s$genes$Symbol
length(unique(genes_sgrna))
nsgrna <- table(table(genes_sgrna))
nsgrna
barplot(nsgrna, col = "#FF9999", ylim=c(0,7000), border = NA, main="The distribution of number of sgRNAs per gene \n (After strict filtering)")
```

```{r library sizes after strict filtering}
yy <- d.filtered.s[, order(d.filtered.s$samples$group, colSums(d.filtered.s$counts))]
yy.logcpm <- cpm(yy$counts, log = T)
boxplot(yy.logcpm, las = 2, pch = 19, cex.axis = 0.5, col = col[as.factor(yy$samples$group)], outcex = 0.3, outcol = col[as.factor(yy$samples$group)],labels = colnames(yy.logcpm), main = "SgRNA abundance after strict filtering", ylab = "SgRNA representation (log2 cpm normalised reads)")
legend("topleft", legend = levels(factor(yy$samples$group)), col = col, pch = 19, cex = 0.5)
```

### Permissive filtering{-}
```{r permissive filtering}
th1 <- as.vector(cpm(1, mean(d$samples$lib.size)))
th1 # average count per million reads (1e6/mean(d$samples$lib.size))
th2 <- 2 # average number of sample per condition
isexprs <- rowSums(cpm(d) > th1) >= th2
table(isexprs)
dim(d.filtered.p <- d[isexprs, ])
```
```{r permissive filtering guide distribution}
genes_sgrna <- d.filtered.p$genes$Symbol
length(unique(genes_sgrna))
nsgrna <- table(table(genes_sgrna))
nsgrna
barplot(nsgrna, col = "#FF9999", ylim=c(0,7000), border = NA, main="The distribution of number of sgRNAs per gene \n (After permissive filtering)")
```

```{r library sizes after permissive filtering}
yy <- d.filtered.p[, order(d.filtered.p$samples$group, colSums(d.filtered.p$counts))]
yy.logcpm <- cpm(yy$counts, log = T)
boxplot(yy.logcpm, las = 2, pch = 19, cex.axis = 0.5, col = col[as.factor(yy$samples$group)], outcex = 0.3, outcol = col[as.factor(yy$samples$group)],labels = colnames(yy.logcpm), main = "SgRNA abundance after permissive filtering", ylab = "SgRNA representation (log2 cpm normalised reads)")
legend("topleft", legend = levels(factor(yy$samples$group)), col = col, pch = 19, cex = 0.5)
```

## Normalising the sgRNA counts{-}
During the sample preparation or sequencing process, external factors that are not of biological interest can affect the expression of individual samples. 
For example, samples processed in the first batch of an experiment can have higher expression overall when compared to samples processed in a second batch. 
It is assumed that all samples should have a similar range and distribution of expression values. 
Normalisation is required to ensure that the expression distributions of each sample are similar across the entire experiment.
Here we applied the upperquartile method from calcNormFactors function in edgeR. In the upper-quartile normalization method of Bullard et al (2010), **Bullard JH, Purdom E, Hansen KD, Dudoit S. (2010) Evaluation of statistical methods for normalization and differential expression in mRNA-Seq experiments. BMC Bioinformatics 11, 94.**, the scale factors are calculated from the 75% quantile of the counts for each library, after removing genes that are zero in all libraries.
When working with DGEList-objects, the normalisation factors are automatically stored. 
For this dataset, the effect of upperquartile-normalisation is mild, as evident in the magnitude of the scaling factors, which are all not too far from 1 as we expected them to be.
```{r normalising}
yy.norm <- d.filtered.s
yy.norm$samples$norm.factors <- calcNormFactors(yy.norm$counts, method = "TMM")
yy.norm$samples[,1:3] # TMM norm factors of strictly filtered data
yy.norm$samples$norm.factors <- calcNormFactors(yy.norm$counts+100, method = "upperquartile")
yy.norm$samples[,1:3] # Upperquartile norm factors of strict filtered data

yy.norm <- d.filtered.p
yy.norm$samples$norm.factors <- calcNormFactors(yy.norm$counts, method = "TMM")
yy.norm$samples[,1:3] # TMM norm factors of permissive filtered data
yy.norm$samples$norm.factors <- calcNormFactors(yy.norm$counts+100, method = "upperquartile")
yy.norm$samples[,1:3] # Upperquartile norm factors of permissive filtered data
```

## MDS plots{-}

```{r mds plot normalisation}
cols <- col
plotMDS(yy.norm, col=cols[factor(yy.norm$samples$group)], main="MDS plot before normalisation", pch = 16, cex = 2)
legend("bottomright", legend=levels(factor(yy.norm$samples$group)), col = cols, pch = 16, cex=0.8)
```

# Differential abundance analysis at guide level{-}
## Create a design matrix{-}
```{r design}
design <- model.matrix(~group, data = yy.norm$samples)
design
```

## Biological variation{-}
```{r bcv}
yy.norm <- estimateDisp(yy.norm, design)
plotBCV(yy.norm)
```

## Fit a statistical model{-}
```{r dea}
fit <- glmFit(yy.norm, design)
lrt <- glmLRT(fit, coef = "groupToxA")
topToxAvsCtrl.all <- topTags(lrt, n = Inf, sort.by = "PValue")$table
topToxAvsCtrl <- topTags(lrt, n = Inf, sort.by = "PValue")$table
dim(topToxAvsCtrl <- topToxAvsCtrl[topToxAvsCtrl$FDR < 0.05,])
topToxAvsCtrl$sgRNA <- rownames(topToxAvsCtrl)
topToxAvsCtrl$Gene_Regulation[topToxAvsCtrl$logFC >= 0] <- "Enriched in ToxA"
topToxAvsCtrl$Gene_Regulation[topToxAvsCtrl$logFC < 0] <- "Depleted in ToxA"
head(topToxAvsCtrl <- topToxAvsCtrl[topToxAvsCtrl$Symbol %in% alias2Symbol(topToxAvsCtrl$Symbol),])
datatable(format(topToxAvsCtrl <- topToxAvsCtrl[order(topToxAvsCtrl$FDR, decreasing = FALSE),c(-2,-3)],format="e", digits=3))
dir.create("tables")
write.csv(topToxAvsCtrl, file = paste0("./tables/",Sys.Date(),"-ToxAvsCtrl-DE-sgRNAs-GGiner.csv"))
```

## Visualise hit guides{-}
A volcano plot is a scatter plot that visualizes the differential abundance of guides. The fold change is typically displayed on the x-axis, while the y-axis represents the corresponding p-values. The significantly differentially abundant genes are the ones found in the upper-left and upper-right corners.
```{r volcano of guides, fig.height=8, fig.width=10}
# remove rows that contain NA values
de <- topToxAvsCtrl.all[complete.cases(topToxAvsCtrl.all), ]
colnames(de)[colnames(de)=="logFC"] <- "log2FoldChange"
colnames(de)[colnames(de)=="PValue"] <- "pvalue"
colnames(de)[colnames(de)=="Symbol"] <- "gene_symbol"
colnames(de)[colnames(de)=="FDR"] <- "adjusted_pvalue"
de$diffabundant <- "Pass p-value cuttoff"
de$diffabundant[de$adjusted_pvalue > 0.05] <- "Not Significant"
de$diffabundant[de$log2FoldChange < -5 & de$log2FoldChange >= 5 & de$adjusted_pvalue < 0.05] <- "Pass p-value & Log2FC cuttoff"
de$diffabundant[de$log2FoldChange >= 5 & de$adjusted_pvalue <= 0.05] <- "Up"
de$diffabundant[de$log2FoldChange < -5 & de$adjusted_pvalue <= 0.05] <- "Down"
head(de)

# Finally, we can organize the labels nicely using the "ggrepel" package and the geom_text_repel() function
g_down <- which(de$log2FoldChange > 5 & de$adjusted_pvalue <= 0.05)
g_up <- which(de$log2FoldChange < -5 & de$adjusted_pvalue <= 0.05)
de$delabel<- NA
de$delabel[c(g_down[1:20],g_up[1:20])] <- de$gene_symbol[c(g_down[1:20],g_up[1:20])]

# plot adding up all layers we have seen so far
ggplot(data=de, aes(x=log2FoldChange, y=-log10(adjusted_pvalue), col=diffabundant, label=delabel)) +
        geom_point() + 
        theme_minimal() +
        geom_text_repel(max.overlaps=Inf) +
        scale_color_manual(values=c("blue", "black", "gray", "red"))+
        geom_vline(xintercept=c(-5, 5), col="darkred") +
        geom_hline(yintercept=-log10(0.05), col="darkred")
```

# Differential abundance analysis at gene level{-}
```{r de genes}
# Create a list of genes with sgRNA indices to use in fry test index argument
genesymbols <- yy.norm$genes[,1]
genesymbollist <- list()
unq <- unique(genesymbols)
unq <- unq[!is.na(unq)]
for(i in unq) {
  sel <- genesymbols==i & !is.na(genesymbols)
  if(sum(sel)>3) genesymbollist[[i]] <- which(sel)
}

fry.res <- fry(yy.norm, index=genesymbollist, design, contrast="groupToxA")
datatable(format(fry.res,format="e", digits=3))
```



