<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Counting the sgRNAs with RSubread | Analysing CRISPR Screens with edgeR</title>
  <meta name="description" content="<p>This is a book version to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Counting the sgRNAs with RSubread | Analysing CRISPR Screens with edgeR" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a book version to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Counting the sgRNAs with RSubread | Analysing CRISPR Screens with edgeR" />
  
  <meta name="twitter:description" content="<p>This is a book version to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  

<meta name="author" content="Göknur Giner" />


<meta name="date" content="2023-10-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="setting-up.html"/>
<link rel="next" href="preprocessing-the-data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.30/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#what-you-will-learn"><i class="fa fa-check"></i>What You Will Learn</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#what-you-wont-learn"><i class="fa fa-check"></i>What You Won’t Learn</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#who-we-wrote-this-for"><i class="fa fa-check"></i>Who We Wrote This For</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#why-we-wrote-this"><i class="fa fa-check"></i>Why We Wrote This</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html"><i class="fa fa-check"></i>Setting up</a>
<ul>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#install-r"><i class="fa fa-check"></i>Install R</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#install-rstudio"><i class="fa fa-check"></i>Install RStudio</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#install-required-packages"><i class="fa fa-check"></i>Install required packages</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#load-required-packages"><i class="fa fa-check"></i>Load required packages</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#download-the-material-used-in-this-book"><i class="fa fa-check"></i>Download the material used in this book</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html"><i class="fa fa-check"></i>Counting the sgRNAs with RSubread</a>
<ul>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html#building-an-index"><i class="fa fa-check"></i>Building an index</a></li>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html#aligning-the-fastq-files"><i class="fa fa-check"></i>Aligning the fastq files</a></li>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html#creating-a-dgelist-object"><i class="fa fa-check"></i>Creating a DGEList object</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html"><i class="fa fa-check"></i>Preprocessing the data</a>
<ul>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#guide-distribution-across-samples"><i class="fa fa-check"></i>Guide distribution across samples</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#gini-indices"><i class="fa fa-check"></i>Gini indices</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#library-sizes-by-group"><i class="fa fa-check"></i>Library sizes by group</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#removing-lowly-expressed-guides"><i class="fa fa-check"></i>Removing lowly expressed guides</a>
<ul>
<li class="chapter" data-level="0.0.1" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#remove-non-targeting-guides"><i class="fa fa-check"></i><b>0.0.1</b> Remove Non-targeting guides</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#strict-filtering"><i class="fa fa-check"></i>Strict filtering</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#permissive-filtering"><i class="fa fa-check"></i>Permissive filtering</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#normalising-the-sgrna-counts"><i class="fa fa-check"></i>Normalising the sgRNA counts</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#mds-plots"><i class="fa fa-check"></i>MDS plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html"><i class="fa fa-check"></i>Differential abundance analysis at guide level</a>
<ul>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#create-a-design-matrix"><i class="fa fa-check"></i>Create a design matrix</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#biological-variation"><i class="fa fa-check"></i>Biological variation</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#fit-a-statistical-model"><i class="fa fa-check"></i>Fit a statistical model</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#visualise-hit-guides"><i class="fa fa-check"></i>Visualise hit guides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-gene-level.html"><a href="differential-abundance-analysis-at-gene-level.html"><i class="fa fa-check"></i>Differential abundance analysis at gene level</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysing CRISPR Screens with edgeR</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="counting-the-sgrnas-with-rsubread" class="section level1 unnumbered hasAnchor">
<h1>Counting the sgRNAs with RSubread<a href="counting-the-sgrnas-with-rsubread.html#counting-the-sgrnas-with-rsubread" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Then we load the guide library, and subsequently, we must convert it into the FASTA format to construct an index for alignment.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="counting-the-sgrnas-with-rsubread.html#cb3-1" tabindex="-1"></a>GeCKO <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;sgRNA_library/GeCKOv21_Human.tsv&quot;</span>)</span>
<span id="cb3-2"><a href="counting-the-sgrnas-with-rsubread.html#cb3-2" tabindex="-1"></a>GeCKO[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, ]</span></code></pre></div>
<pre><code>  gene_id          UID                  seq
1    A1BG HGLibA_00001 GTCGCTGAGCTCCGATTCGA
2    A1BG HGLibA_00002 ACCTGTAGTTGCCGGCGTGC</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="counting-the-sgrnas-with-rsubread.html#cb5-1" tabindex="-1"></a>sgRNAs <span class="ot">&lt;-</span> <span class="fu">DNAStringSet</span>(GeCKO<span class="sc">$</span>seq)</span>
<span id="cb5-2"><a href="counting-the-sgrnas-with-rsubread.html#cb5-2" tabindex="-1"></a><span class="fu">names</span>(sgRNAs) <span class="ot">&lt;-</span> GeCKO<span class="sc">$</span>UID</span>
<span id="cb5-3"><a href="counting-the-sgrnas-with-rsubread.html#cb5-3" tabindex="-1"></a>sgRNAs</span></code></pre></div>
<pre><code>DNAStringSet object of length 122411:
         width seq                                          names               
     [1]    20 GTCGCTGAGCTCCGATTCGA                         HGLibA_00001
     [2]    20 ACCTGTAGTTGCCGGCGTGC                         HGLibA_00002
     [3]    20 CGTCAGCGTCACATTGGCCA                         HGLibA_00003
     [4]    20 CGCGCACTGGTCCAGCGCAC                         HGLibA_00004
     [5]    20 CCAAGCTATATCCTGTGCGC                         HGLibA_00005
     ...   ... ...
[122407]    20 GGTGATGCCCTACCCGATGC                         HGLibB_57024
[122408]    20 GAGGGCCTCCAACATGTTCT                         HGLibB_57025
[122409]    20 GTTCTTCAACAGTCCACAAC                         HGLibB_57026
[122410]    20 CGTCGAGATTCTACTTCTTC                         HGLibB_57027
[122411]    20 CTGAAACATTTAACCAGTTG                         HGLibB_57028</code></pre>
<div id="building-an-index" class="section level2 unnumbered hasAnchor">
<h2>Building an index<a href="counting-the-sgrnas-with-rsubread.html#building-an-index" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="counting-the-sgrnas-with-rsubread.html#cb7-1" tabindex="-1"></a><span class="fu">buildindex</span>(<span class="st">&quot;./index/GeCKO&quot;</span>, <span class="st">&quot;./index/GeCKO.fa&quot;</span>, <span class="at">indexSplit =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="aligning-the-fastq-files" class="section level2 unnumbered hasAnchor">
<h2>Aligning the fastq files<a href="counting-the-sgrnas-with-rsubread.html#aligning-the-fastq-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="counting-the-sgrnas-with-rsubread.html#cb8-1" tabindex="-1"></a>fastqs <span class="ot">&lt;-</span> <span class="fu">dir</span>(<span class="at">path =</span> <span class="st">&quot;./fastq_files&quot;</span>, <span class="at">pattern =</span> <span class="st">&quot;*.fastq.gz&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-2"><a href="counting-the-sgrnas-with-rsubread.html#cb8-2" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb8-3"><a href="counting-the-sgrnas-with-rsubread.html#cb8-3" tabindex="-1"></a>mapping_results <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb8-4"><a href="counting-the-sgrnas-with-rsubread.html#cb8-4" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fastqs)) {</span>
<span id="cb8-5"><a href="counting-the-sgrnas-with-rsubread.html#cb8-5" tabindex="-1"></a>    mapping_results[[i]] <span class="ot">&lt;-</span> <span class="fu">align</span>(<span class="st">&quot;./index/GeCKO&quot;</span>, fastqs[i], <span class="at">output_file =</span> <span class="fu">gsub</span>(<span class="st">&quot;.fastq.gz&quot;</span>,</span>
<span id="cb8-6"><a href="counting-the-sgrnas-with-rsubread.html#cb8-6" tabindex="-1"></a>        <span class="st">&quot;.bam&quot;</span>, fastqs[i]), <span class="at">nthreads =</span> <span class="dv">4</span>, <span class="at">unique =</span> <span class="cn">TRUE</span>, <span class="at">nBestLocations =</span> <span class="dv">1</span>, <span class="at">type =</span> <span class="st">&quot;DNA&quot;</span>,</span>
<span id="cb8-7"><a href="counting-the-sgrnas-with-rsubread.html#cb8-7" tabindex="-1"></a>        <span class="at">TH1 =</span> <span class="dv">1</span>, <span class="at">maxMismatches =</span> <span class="dv">0</span>, <span class="at">indels =</span> <span class="dv">0</span>)</span>
<span id="cb8-8"><a href="counting-the-sgrnas-with-rsubread.html#cb8-8" tabindex="-1"></a>    temp <span class="ot">&lt;-</span> <span class="fu">readGAlignments</span>(<span class="fu">gsub</span>(<span class="st">&quot;.fastq.gz&quot;</span>, <span class="st">&quot;.bam&quot;</span>, fastqs[i]))</span>
<span id="cb8-9"><a href="counting-the-sgrnas-with-rsubread.html#cb8-9" tabindex="-1"></a>    counts[[i]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">table</span>(<span class="fu">seqnames</span>(temp[<span class="fu">width</span>(temp) <span class="sc">==</span> <span class="st">&quot;20&quot;</span>])), <span class="at">row.names =</span> <span class="st">&quot;Var1&quot;</span>)</span>
<span id="cb8-10"><a href="counting-the-sgrnas-with-rsubread.html#cb8-10" tabindex="-1"></a>}</span>
<span id="cb8-11"><a href="counting-the-sgrnas-with-rsubread.html#cb8-11" tabindex="-1"></a>my_counts <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, counts)</span>
<span id="cb8-12"><a href="counting-the-sgrnas-with-rsubread.html#cb8-12" tabindex="-1"></a><span class="fu">colnames</span>(my_counts) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Control_1&quot;</span>, <span class="st">&quot;Control_2&quot;</span>, <span class="st">&quot;ToxA_1&quot;</span>, <span class="st">&quot;ToxA_2&quot;</span>, <span class="st">&quot;ToxB_1&quot;</span>,</span>
<span id="cb8-13"><a href="counting-the-sgrnas-with-rsubread.html#cb8-13" tabindex="-1"></a>    <span class="st">&quot;ToxB_2&quot;</span>)</span></code></pre></div>
</div>
<div id="creating-a-dgelist-object" class="section level2 unnumbered hasAnchor">
<h2>Creating a DGEList object<a href="counting-the-sgrnas-with-rsubread.html#creating-a-dgelist-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="counting-the-sgrnas-with-rsubread.html#cb9-1" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;my_counts.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-2"><a href="counting-the-sgrnas-with-rsubread.html#cb9-2" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">&quot;Control&quot;</span>, <span class="st">&quot;Control&quot;</span>, <span class="st">&quot;ToxA&quot;</span>, <span class="st">&quot;ToxA&quot;</span>, <span class="st">&quot;ToxB&quot;</span>, <span class="st">&quot;ToxB&quot;</span>), <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span>,</span>
<span id="cb9-3"><a href="counting-the-sgrnas-with-rsubread.html#cb9-3" tabindex="-1"></a>    <span class="st">&quot;ToxA&quot;</span>, <span class="st">&quot;ToxB&quot;</span>))</span>
<span id="cb9-4"><a href="counting-the-sgrnas-with-rsubread.html#cb9-4" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">group =</span> group, <span class="at">sampleName =</span> <span class="fu">colnames</span>(counts), <span class="at">biorep =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="dv">1</span>,</span>
<span id="cb9-5"><a href="counting-the-sgrnas-with-rsubread.html#cb9-5" tabindex="-1"></a>    <span class="dv">2</span>), <span class="dv">3</span>))</span>
<span id="cb9-6"><a href="counting-the-sgrnas-with-rsubread.html#cb9-6" tabindex="-1"></a>genes <span class="ot">&lt;-</span> GeCKO</span>
<span id="cb9-7"><a href="counting-the-sgrnas-with-rsubread.html#cb9-7" tabindex="-1"></a><span class="fu">names</span>(genes)[<span class="fu">names</span>(genes) <span class="sc">==</span> <span class="st">&quot;gene_id&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Symbol&quot;</span></span>
<span id="cb9-8"><a href="counting-the-sgrnas-with-rsubread.html#cb9-8" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts =</span> counts, <span class="at">samples =</span> samples, <span class="at">genes =</span> genes)</span>
<span id="cb9-9"><a href="counting-the-sgrnas-with-rsubread.html#cb9-9" tabindex="-1"></a>d</span></code></pre></div>
<pre><code>An object of class &quot;DGEList&quot;
$counts
             Control_1 Control_2 ToxA_1 ToxA_2 ToxB_1 ToxB_2
HGLibA_00001         0         7      3      2      2      8
HGLibA_00002         0         0      0      0      0      0
HGLibA_00003         1         0      0      0      0      0
HGLibA_00004         9         0      0      0      0      1
HGLibA_00005         1         8      1      0      1      1
122406 more rows ...

$samples
            group lib.size norm.factors sampleName biorep
Control_1 Control   338214            1  Control_1      1
Control_2 Control   337711            1  Control_2      2
ToxA_1       ToxA   344188            1     ToxA_1      1
ToxA_2       ToxA   335848            1     ToxA_2      2
ToxB_1       ToxB   356569            1     ToxB_1      1
ToxB_2       ToxB   355131            1     ToxB_2      2

$genes
             Symbol          UID                  seq
HGLibA_00001   A1BG HGLibA_00001 GTCGCTGAGCTCCGATTCGA
HGLibA_00002   A1BG HGLibA_00002 ACCTGTAGTTGCCGGCGTGC
HGLibA_00003   A1BG HGLibA_00003 CGTCAGCGTCACATTGGCCA
HGLibA_00004   A1CF HGLibA_00004 CGCGCACTGGTCCAGCGCAC
HGLibA_00005   A1CF HGLibA_00005 CCAAGCTATATCCTGTGCGC
122406 more rows ...</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="counting-the-sgrnas-with-rsubread.html#cb11-1" tabindex="-1"></a><span class="do">## dir.create(&quot;RData&quot;)</span></span>
<span id="cb11-2"><a href="counting-the-sgrnas-with-rsubread.html#cb11-2" tabindex="-1"></a><span class="fu">save</span>(d, <span class="at">file =</span> <span class="st">&quot;./RData/DGEList.RData&quot;</span>)</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="setting-up.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="preprocessing-the-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
