<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Counting the sgRNAs with RSubread | Analysing CRISPR Screens with edgeR</title>
  <meta name="description" content="<p>This is a book version to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Counting the sgRNAs with RSubread | Analysing CRISPR Screens with edgeR" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a book version to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Counting the sgRNAs with RSubread | Analysing CRISPR Screens with edgeR" />
  
  <meta name="twitter:description" content="<p>This is a book version to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  

<meta name="author" content="GÃ¶knur Giner" />


<meta name="date" content="2023-10-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="setting-up.html"/>
<link rel="next" href="preprocessing-the-data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.30/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Analysing-CRISPR-screens-with-edgeR</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html"><i class="fa fa-check"></i>Setting up</a>
<ul>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#install-r"><i class="fa fa-check"></i>Install R</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#install-rstudio"><i class="fa fa-check"></i>Install RStudio</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#install-required-packages"><i class="fa fa-check"></i>Install required packages</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#load-required-packages"><i class="fa fa-check"></i>Load required packages</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#download-the-material"><i class="fa fa-check"></i>Download the material</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#create-necessary-directories"><i class="fa fa-check"></i>Create necessary directories</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html"><i class="fa fa-check"></i>Counting the sgRNAs with RSubread</a>
<ul>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html#build-an-index"><i class="fa fa-check"></i>Build an index</a></li>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html#align-the-fastq-files"><i class="fa fa-check"></i>Align the FASTQ files</a></li>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html#create-a-dgelist-object"><i class="fa fa-check"></i>Create a DGEList object</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html"><i class="fa fa-check"></i>Preprocessing the data</a>
<ul>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#remove-control-guides"><i class="fa fa-check"></i>Remove control guides</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#check-guide-distribution"><i class="fa fa-check"></i>Check guide distribution</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#evaluate-the-uniformity"><i class="fa fa-check"></i>Evaluate the uniformity</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#remove-lowly-expressed-guides"><i class="fa fa-check"></i>Remove lowly expressed guides</a>
<ul>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#strict-filtering"><i class="fa fa-check"></i>Strict filtering</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#permissive-filtering"><i class="fa fa-check"></i>Permissive filtering</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#look-at-density-plots"><i class="fa fa-check"></i>Look at density plots</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#normalise-the-counts"><i class="fa fa-check"></i>Normalise the counts</a>
<ul>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#cpm-normalised-counts"><i class="fa fa-check"></i>CPM normalised counts</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#tmm-normalised-counts"><i class="fa fa-check"></i>TMM normalised counts</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i>Unsupervised clustering of samples</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#visualise-guide-abundance-profiles-of-samples"><i class="fa fa-check"></i>Visualise guide abundance profiles of samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html"><i class="fa fa-check"></i>Differential abundance analysis at guide level</a>
<ul>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#create-the-design-matrix"><i class="fa fa-check"></i>Create the design matrix</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#fit-a-model-when-there-is-no-biological-replicates"><i class="fa fa-check"></i>Fit a model when there is no biological replicates</a>
<ul>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#classic-edger"><i class="fa fa-check"></i>Classic edgeR</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#edger-glmlrt"><i class="fa fa-check"></i>edgeR glmLRT</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#fit-a-model-when-there-are-biological-replicates"><i class="fa fa-check"></i>Fit a model when there are biological replicates</a>
<ul>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#dispersion-estimates-and-biological-variation"><i class="fa fa-check"></i>Dispersion estimates and biological variation</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#edger-glmlrt-1"><i class="fa fa-check"></i>edgeR glmLRT</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#edger-quasi"><i class="fa fa-check"></i>edgeR quasi</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="visualise-hit-guides.html"><a href="visualise-hit-guides.html"><i class="fa fa-check"></i>Visualise hit guides</a>
<ul>
<li class="chapter" data-level="" data-path="visualise-hit-guides.html"><a href="visualise-hit-guides.html#volcano-plot-toxa-vs-control"><i class="fa fa-check"></i>Volcano plot (ToxA vs Control)</a></li>
<li class="chapter" data-level="" data-path="visualise-hit-guides.html"><a href="visualise-hit-guides.html#volcano-plot-toxb-vs-control"><i class="fa fa-check"></i>Volcano plot (ToxB vs Control)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-gene-level.html"><a href="differential-abundance-analysis-at-gene-level.html"><i class="fa fa-check"></i>Differential abundance analysis at gene level</a>
<ul>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-gene-level.html"><a href="differential-abundance-analysis-at-gene-level.html#toxa-vs-control"><i class="fa fa-check"></i>ToxA vs Control</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-gene-level.html"><a href="differential-abundance-analysis-at-gene-level.html#toxb-vs-control"><i class="fa fa-check"></i>ToxB vs Control</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysing CRISPR Screens with edgeR</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="counting-the-sgrnas-with-rsubread" class="section level1 unnumbered hasAnchor">
<h1>Counting the sgRNAs with RSubread<a href="counting-the-sgrnas-with-rsubread.html#counting-the-sgrnas-with-rsubread" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>To count the sgRNAs, we must first load the human GeCKO sgRNA library, and subsequently, convert it into the FASTA format to construct an index for alignment. To do that we use <code>DNAStringSet</code> and <code>writeXStringSet</code> functions from <code>GenomicAlignments</code> Bioconductor package.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="counting-the-sgrnas-with-rsubread.html#cb4-1" tabindex="-1"></a>GeCKO <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;sgRNA_library/GeCKOv21_Human.tsv&quot;</span>)</span>
<span id="cb4-2"><a href="counting-the-sgrnas-with-rsubread.html#cb4-2" tabindex="-1"></a>GeCKO[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, ]</span></code></pre></div>
<pre><code>  gene_id          UID                  seq
1    A1BG HGLibA_00001 GTCGCTGAGCTCCGATTCGA
2    A1BG HGLibA_00002 ACCTGTAGTTGCCGGCGTGC</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="counting-the-sgrnas-with-rsubread.html#cb6-1" tabindex="-1"></a>sgRNAs <span class="ot">&lt;-</span> <span class="fu">DNAStringSet</span>(GeCKO<span class="sc">$</span>seq)</span>
<span id="cb6-2"><a href="counting-the-sgrnas-with-rsubread.html#cb6-2" tabindex="-1"></a><span class="fu">names</span>(sgRNAs) <span class="ot">&lt;-</span> GeCKO<span class="sc">$</span>UID</span>
<span id="cb6-3"><a href="counting-the-sgrnas-with-rsubread.html#cb6-3" tabindex="-1"></a>sgRNAs</span></code></pre></div>
<pre><code>DNAStringSet object of length 122411:
         width seq                                        names               
     [1]    20 [48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m[48;5;217m[30mA[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;217m[30mA[39m[49m[48;5;223m[30mT[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;217m[30mA[39m[49m                       HGLibA_00001
     [2]    20 [48;5;217m[30mA[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;217m[30mA[39m[49m[48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m                       HGLibA_00002
     [3]    20 [48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;223m[30mT[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m                       HGLibA_00003
     [4]    20 [48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m[48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;157m[30mC[39m[49m                       HGLibA_00004
     [5]    20 [48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;217m[30mA[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m[48;5;217m[30mA[39m[49m[48;5;223m[30mT[39m[49m[48;5;217m[30mA[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m                       HGLibA_00005
     ...   ... ...
[122407]    20 [48;5;159m[30mG[39m[49m[48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m[48;5;217m[30mA[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m[48;5;217m[30mA[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;217m[30mA[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m                       HGLibB_57024
[122408]    20 [48;5;159m[30mG[39m[49m[48;5;217m[30mA[39m[49m[48;5;159m[30mG[39m[49m[48;5;159m[30mG[39m[49m[48;5;159m[30mG[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;217m[30mA[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m                       HGLibB_57025
[122409]    20 [48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;217m[30mA[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;217m[30mA[39m[49m[48;5;157m[30mC[39m[49m                       HGLibB_57026
[122410]    20 [48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;159m[30mG[39m[49m[48;5;217m[30mA[39m[49m[48;5;159m[30mG[39m[49m[48;5;217m[30mA[39m[49m[48;5;223m[30mT[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m[48;5;217m[30mA[39m[49m[48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m[48;5;223m[30mT[39m[49m[48;5;157m[30mC[39m[49m                       HGLibB_57027
[122411]    20 [48;5;157m[30mC[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m[48;5;217m[30mA[39m[49m[48;5;217m[30mA[39m[49m[48;5;217m[30mA[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;223m[30mT[39m[49m[48;5;223m[30mT[39m[49m[48;5;223m[30mT[39m[49m[48;5;217m[30mA[39m[49m[48;5;217m[30mA[39m[49m[48;5;157m[30mC[39m[49m[48;5;157m[30mC[39m[49m[48;5;217m[30mA[39m[49m[48;5;159m[30mG[39m[49m[48;5;223m[30mT[39m[49m[48;5;223m[30mT[39m[49m[48;5;159m[30mG[39m[49m                       HGLibB_57028</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="counting-the-sgrnas-with-rsubread.html#cb8-1" tabindex="-1"></a><span class="fu">writeXStringSet</span>(sgRNAs, <span class="at">file =</span> <span class="st">&quot;./index/GeCKO.fa&quot;</span>)</span></code></pre></div>
<div id="build-an-index" class="section level2 unnumbered hasAnchor">
<h2>Build an index<a href="counting-the-sgrnas-with-rsubread.html#build-an-index" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Then we build an index of sgRNA library with <code>buildindex</code> function from <code>RSubread</code> package.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="counting-the-sgrnas-with-rsubread.html#cb9-1" tabindex="-1"></a><span class="fu">buildindex</span>(<span class="st">&quot;./index/GeCKO&quot;</span>, <span class="st">&quot;./index/GeCKO.fa&quot;</span>, <span class="at">indexSplit =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="align-the-fastq-files" class="section level2 unnumbered hasAnchor">
<h2>Align the FASTQ files<a href="counting-the-sgrnas-with-rsubread.html#align-the-fastq-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next step is to align the sequencing files with <code>align</code> function from <code>RSubread</code> and count the reads with <code>readGAlignments</code> function in <code>GenomicAlignments</code>.</p>
<p>The subsequent phase involves aligning the sequencing files (<a href="https://github.com/goknurginer/Analysing-CRISPR-screens-with-edgeR/tree/main/fasqfiles">FASTQ files</a>) using the <code>align</code> function from <code>RSubread</code> and counting the reads for each sgRNA using the <code>readGAlignments</code> function in <code>GenomicAlignments</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="counting-the-sgrnas-with-rsubread.html#cb10-1" tabindex="-1"></a>fastqs <span class="ot">&lt;-</span> <span class="fu">dir</span>(<span class="at">path =</span> <span class="st">&quot;./fastq_files&quot;</span>, <span class="at">pattern =</span> <span class="st">&quot;*.fastq.gz&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-2"><a href="counting-the-sgrnas-with-rsubread.html#cb10-2" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb10-3"><a href="counting-the-sgrnas-with-rsubread.html#cb10-3" tabindex="-1"></a>mapping_results <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb10-4"><a href="counting-the-sgrnas-with-rsubread.html#cb10-4" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fastqs)) {</span>
<span id="cb10-5"><a href="counting-the-sgrnas-with-rsubread.html#cb10-5" tabindex="-1"></a>    mapping_results[[i]] <span class="ot">&lt;-</span> <span class="fu">align</span>(<span class="st">&quot;./index/GeCKO&quot;</span>, fastqs[i], <span class="at">output_file =</span> <span class="fu">gsub</span>(<span class="st">&quot;.fastq.gz&quot;</span>,</span>
<span id="cb10-6"><a href="counting-the-sgrnas-with-rsubread.html#cb10-6" tabindex="-1"></a>        <span class="st">&quot;.bam&quot;</span>, fastqs[i]), <span class="at">nthreads =</span> <span class="dv">4</span>, <span class="at">unique =</span> <span class="cn">TRUE</span>, <span class="at">nBestLocations =</span> <span class="dv">1</span>, <span class="at">type =</span> <span class="st">&quot;DNA&quot;</span>,</span>
<span id="cb10-7"><a href="counting-the-sgrnas-with-rsubread.html#cb10-7" tabindex="-1"></a>        <span class="at">TH1 =</span> <span class="dv">1</span>, <span class="at">maxMismatches =</span> <span class="dv">0</span>, <span class="at">indels =</span> <span class="dv">0</span>)</span>
<span id="cb10-8"><a href="counting-the-sgrnas-with-rsubread.html#cb10-8" tabindex="-1"></a>    temp <span class="ot">&lt;-</span> <span class="fu">readGAlignments</span>(<span class="fu">gsub</span>(<span class="st">&quot;.fastq.gz&quot;</span>, <span class="st">&quot;.bam&quot;</span>, fastqs[i]))</span>
<span id="cb10-9"><a href="counting-the-sgrnas-with-rsubread.html#cb10-9" tabindex="-1"></a>    counts[[i]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">table</span>(<span class="fu">seqnames</span>(temp[<span class="fu">width</span>(temp) <span class="sc">==</span> <span class="st">&quot;20&quot;</span>])), <span class="at">row.names =</span> <span class="st">&quot;Var1&quot;</span>)</span>
<span id="cb10-10"><a href="counting-the-sgrnas-with-rsubread.html#cb10-10" tabindex="-1"></a>}</span>
<span id="cb10-11"><a href="counting-the-sgrnas-with-rsubread.html#cb10-11" tabindex="-1"></a>my_counts <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, counts)</span>
<span id="cb10-12"><a href="counting-the-sgrnas-with-rsubread.html#cb10-12" tabindex="-1"></a><span class="fu">colnames</span>(my_counts) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Control_1&quot;</span>, <span class="st">&quot;Control_2&quot;</span>, <span class="st">&quot;ToxA_1&quot;</span>, <span class="st">&quot;ToxA_2&quot;</span>, <span class="st">&quot;ToxB_1&quot;</span>,</span>
<span id="cb10-13"><a href="counting-the-sgrnas-with-rsubread.html#cb10-13" tabindex="-1"></a>    <span class="st">&quot;ToxB_2&quot;</span>)</span>
<span id="cb10-14"><a href="counting-the-sgrnas-with-rsubread.html#cb10-14" tabindex="-1"></a><span class="fu">write.table</span>(my_counts, <span class="st">&quot;my_counts.txt&quot;</span>)</span></code></pre></div>
</div>
<div id="create-a-dgelist-object" class="section level2 unnumbered hasAnchor">
<h2>Create a DGEList object<a href="counting-the-sgrnas-with-rsubread.html#create-a-dgelist-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For downstream analysis, here we are going to convert count matrix obtained in the previous section into a DGEList object using the <code>DGEList</code> function from <code>edgeR</code> package. The DGEList object consists of three components: counts, information about samples and gene annotations.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="counting-the-sgrnas-with-rsubread.html#cb11-1" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;my_counts.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-2"><a href="counting-the-sgrnas-with-rsubread.html#cb11-2" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">&quot;Control&quot;</span>, <span class="st">&quot;Control&quot;</span>, <span class="st">&quot;ToxA&quot;</span>, <span class="st">&quot;ToxA&quot;</span>, <span class="st">&quot;ToxB&quot;</span>, <span class="st">&quot;ToxB&quot;</span>), <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span>,</span>
<span id="cb11-3"><a href="counting-the-sgrnas-with-rsubread.html#cb11-3" tabindex="-1"></a>    <span class="st">&quot;ToxA&quot;</span>, <span class="st">&quot;ToxB&quot;</span>))</span>
<span id="cb11-4"><a href="counting-the-sgrnas-with-rsubread.html#cb11-4" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">group =</span> group, <span class="at">sampleName =</span> <span class="fu">colnames</span>(counts), <span class="at">biorep =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="dv">1</span>,</span>
<span id="cb11-5"><a href="counting-the-sgrnas-with-rsubread.html#cb11-5" tabindex="-1"></a>    <span class="dv">2</span>), <span class="dv">3</span>))</span>
<span id="cb11-6"><a href="counting-the-sgrnas-with-rsubread.html#cb11-6" tabindex="-1"></a>genes <span class="ot">&lt;-</span> GeCKO</span>
<span id="cb11-7"><a href="counting-the-sgrnas-with-rsubread.html#cb11-7" tabindex="-1"></a><span class="fu">names</span>(genes)[<span class="fu">names</span>(genes) <span class="sc">==</span> <span class="st">&quot;gene_id&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Symbol&quot;</span></span>
<span id="cb11-8"><a href="counting-the-sgrnas-with-rsubread.html#cb11-8" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts =</span> counts, <span class="at">samples =</span> samples, <span class="at">genes =</span> genes)</span>
<span id="cb11-9"><a href="counting-the-sgrnas-with-rsubread.html#cb11-9" tabindex="-1"></a>d</span></code></pre></div>
<pre><code>An object of class &quot;DGEList&quot;
$counts
             Control_1 Control_2 ToxA_1 ToxA_2 ToxB_1 ToxB_2
HGLibA_00001         0         7      3      2      2      8
HGLibA_00002         0         0      0      0      0      0
HGLibA_00003         1         0      0      0      0      0
HGLibA_00004         9         0      0      0      0      1
HGLibA_00005         1         8      1      0      1      1
122406 more rows ...

$samples
            group lib.size norm.factors sampleName biorep
Control_1 Control   338214            1  Control_1      1
Control_2 Control   337711            1  Control_2      2
ToxA_1       ToxA   344188            1     ToxA_1      1
ToxA_2       ToxA   335848            1     ToxA_2      2
ToxB_1       ToxB   356569            1     ToxB_1      1
ToxB_2       ToxB   355131            1     ToxB_2      2

$genes
             Symbol          UID                  seq
HGLibA_00001   A1BG HGLibA_00001 GTCGCTGAGCTCCGATTCGA
HGLibA_00002   A1BG HGLibA_00002 ACCTGTAGTTGCCGGCGTGC
HGLibA_00003   A1BG HGLibA_00003 CGTCAGCGTCACATTGGCCA
HGLibA_00004   A1CF HGLibA_00004 CGCGCACTGGTCCAGCGCAC
HGLibA_00005   A1CF HGLibA_00005 CCAAGCTATATCCTGTGCGC
122406 more rows ...</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="counting-the-sgrnas-with-rsubread.html#cb13-1" tabindex="-1"></a><span class="fu">save</span>(d, <span class="at">file =</span> <span class="st">&quot;./RData/DGEList.RData&quot;</span>)</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="setting-up.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="preprocessing-the-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/goknurginer/Analysing-CRISPR-screens-with-edgeRindex.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Analysing-CRISPR-screens-with-edgeR.pdf", "Analysing-CRISPR-screens-with-edgeR.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
