<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Preprocessing the data | Analysing CRISPR Screens with edgeR</title>
  <meta name="description" content="<p>This is a book version to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Preprocessing the data | Analysing CRISPR Screens with edgeR" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a book version to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Preprocessing the data | Analysing CRISPR Screens with edgeR" />
  
  <meta name="twitter:description" content="<p>This is a book version to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  

<meta name="author" content="Göknur Giner" />


<meta name="date" content="2023-10-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="counting-the-sgrnas-with-rsubread.html"/>
<link rel="next" href="differential-abundance-analysis-at-guide-level.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.30/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#what-you-will-learn"><i class="fa fa-check"></i>What You Will Learn</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#what-you-wont-learn"><i class="fa fa-check"></i>What You Won’t Learn</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#who-we-wrote-this-for"><i class="fa fa-check"></i>Who We Wrote This For</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#why-we-wrote-this"><i class="fa fa-check"></i>Why We Wrote This</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html"><i class="fa fa-check"></i>Setting up</a>
<ul>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#install-r"><i class="fa fa-check"></i>Install R</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#install-rstudio"><i class="fa fa-check"></i>Install RStudio</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#install-required-packages"><i class="fa fa-check"></i>Install required packages</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#load-required-packages"><i class="fa fa-check"></i>Load required packages</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#download-the-material-used-in-this-book"><i class="fa fa-check"></i>Download the material used in this book</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html"><i class="fa fa-check"></i>Counting the sgRNAs with RSubread</a>
<ul>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html#building-an-index"><i class="fa fa-check"></i>Building an index</a></li>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html#aligning-the-fastq-files"><i class="fa fa-check"></i>Aligning the fastq files</a></li>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html#creating-a-dgelist-object"><i class="fa fa-check"></i>Creating a DGEList object</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html"><i class="fa fa-check"></i>Preprocessing the data</a>
<ul>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#guide-distribution-across-samples"><i class="fa fa-check"></i>Guide distribution across samples</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#gini-indices"><i class="fa fa-check"></i>Gini indices</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#library-sizes-by-group"><i class="fa fa-check"></i>Library sizes by group</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#removing-lowly-expressed-guides"><i class="fa fa-check"></i>Removing lowly expressed guides</a>
<ul>
<li class="chapter" data-level="0.0.1" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#remove-non-targeting-guides"><i class="fa fa-check"></i><b>0.0.1</b> Remove Non-targeting guides</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#strict-filtering"><i class="fa fa-check"></i>Strict filtering</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#permissive-filtering"><i class="fa fa-check"></i>Permissive filtering</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#normalising-the-sgrna-counts"><i class="fa fa-check"></i>Normalising the sgRNA counts</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#mds-plots"><i class="fa fa-check"></i>MDS plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html"><i class="fa fa-check"></i>Differential abundance analysis at guide level</a>
<ul>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#create-a-design-matrix"><i class="fa fa-check"></i>Create a design matrix</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#biological-variation"><i class="fa fa-check"></i>Biological variation</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#fit-a-statistical-model"><i class="fa fa-check"></i>Fit a statistical model</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#visualise-hit-guides"><i class="fa fa-check"></i>Visualise hit guides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-gene-level.html"><a href="differential-abundance-analysis-at-gene-level.html"><i class="fa fa-check"></i>Differential abundance analysis at gene level</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysing CRISPR Screens with edgeR</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="preprocessing-the-data" class="section level1 unnumbered hasAnchor">
<h1>Preprocessing the data<a href="preprocessing-the-data.html#preprocessing-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="preprocessing-the-data.html#cb12-1" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;./RData/DGEList.RData&quot;</span>)</span></code></pre></div>
<div id="guide-distribution-across-samples" class="section level2 unnumbered hasAnchor">
<h2>Guide distribution across samples<a href="preprocessing-the-data.html#guide-distribution-across-samples" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="preprocessing-the-data.html#cb13-1" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb13-2"><a href="preprocessing-the-data.html#cb13-2" tabindex="-1"></a>sample_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(d)</span>
<span id="cb13-3"><a href="preprocessing-the-data.html#cb13-3" tabindex="-1"></a>l <span class="ot">&lt;-</span> <span class="fu">colSums</span>(<span class="sc">!</span>d<span class="sc">$</span>counts <span class="sc">==</span> <span class="dv">0</span>)</span>
<span id="cb13-4"><a href="preprocessing-the-data.html#cb13-4" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">row.name =</span> sample_names, <span class="at">l =</span> l, <span class="at">t =</span> <span class="fu">as.character</span>(d<span class="sc">$</span>samples<span class="sc">$</span>group))</span>
<span id="cb13-5"><a href="preprocessing-the-data.html#cb13-5" tabindex="-1"></a>data <span class="ot">&lt;-</span> data[<span class="fu">order</span>(data<span class="sc">$</span>l, d<span class="sc">$</span>samples<span class="sc">$</span>group), ]</span>
<span id="cb13-6"><a href="preprocessing-the-data.html#cb13-6" tabindex="-1"></a>col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#6DAEB0&quot;</span>, <span class="st">&quot;#FAA578&quot;</span>, <span class="st">&quot;#FFB6C1&quot;</span>)</span>
<span id="cb13-7"><a href="preprocessing-the-data.html#cb13-7" tabindex="-1"></a>my_bar <span class="ot">&lt;-</span> <span class="fu">barplot</span>(data<span class="sc">$</span>l, <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">70000</span>), <span class="at">names.arg =</span> data<span class="sc">$</span>row.name, <span class="at">las =</span> <span class="dv">1</span>,</span>
<span id="cb13-8"><a href="preprocessing-the-data.html#cb13-8" tabindex="-1"></a>    <span class="at">space =</span> <span class="dv">1</span>, <span class="at">xlab =</span> <span class="st">&quot;The number of detected guides&quot;</span>, <span class="at">main =</span> <span class="st">&quot;The distribution of guides&quot;</span>,</span>
<span id="cb13-9"><a href="preprocessing-the-data.html#cb13-9" tabindex="-1"></a>    <span class="at">col =</span> col[<span class="fu">as.factor</span>(data<span class="sc">$</span>t)], <span class="at">border =</span> <span class="cn">NA</span>, <span class="at">legend =</span> <span class="cn">TRUE</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-10"><a href="preprocessing-the-data.html#cb13-10" tabindex="-1"></a><span class="fu">text</span>(my_bar, <span class="at">x =</span> data<span class="sc">$</span>l <span class="sc">+</span> <span class="dv">3000</span>  <span class="co"># adjust this number</span></span>
<span id="cb13-11"><a href="preprocessing-the-data.html#cb13-11" tabindex="-1"></a>, <span class="fu">paste0</span>(<span class="st">&quot;(n:&quot;</span>, data<span class="sc">$</span>l,</span>
<span id="cb13-12"><a href="preprocessing-the-data.html#cb13-12" tabindex="-1"></a>    <span class="st">&quot;)&quot;</span>), <span class="at">las =</span> <span class="dv">2</span>)</span>
<span id="cb13-13"><a href="preprocessing-the-data.html#cb13-13" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;bottomright&quot;</span>, <span class="at">legend =</span> <span class="fu">levels</span>(<span class="fu">factor</span>(data<span class="sc">$</span>t)), <span class="at">fill =</span> col)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/guide%20distribution-1.png" width="576" /></p>
</div>
<div id="gini-indices" class="section level2 unnumbered hasAnchor">
<h2>Gini indices<a href="preprocessing-the-data.html#gini-indices" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="preprocessing-the-data.html#cb14-1" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">gini</span>(d<span class="sc">$</span>counts) <span class="sc">~</span> d<span class="sc">$</span>samples<span class="sc">$</span>sampleName, <span class="at">ylab =</span> <span class="st">&quot;Gini Indices&quot;</span>, <span class="at">xlab =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb14-2"><a href="preprocessing-the-data.html#cb14-2" tabindex="-1"></a>    <span class="at">cex.axis =</span> <span class="fl">0.8</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/gini-1.png" width="576" /></p>
</div>
<div id="library-sizes-by-group" class="section level2 unnumbered hasAnchor">
<h2>Library sizes by group<a href="preprocessing-the-data.html#library-sizes-by-group" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="preprocessing-the-data.html#cb15-1" tabindex="-1"></a>yy <span class="ot">&lt;-</span> d[, <span class="fu">order</span>(d<span class="sc">$</span>samples<span class="sc">$</span>group, <span class="fu">colSums</span>(d<span class="sc">$</span>counts))]</span>
<span id="cb15-2"><a href="preprocessing-the-data.html#cb15-2" tabindex="-1"></a>yy.logcpm <span class="ot">&lt;-</span> <span class="fu">cpm</span>(yy<span class="sc">$</span>counts, <span class="at">log =</span> T)</span>
<span id="cb15-3"><a href="preprocessing-the-data.html#cb15-3" tabindex="-1"></a><span class="fu">boxplot</span>(yy.logcpm, <span class="at">las =</span> <span class="dv">2</span>, <span class="at">pch =</span> <span class="dv">19</span>, <span class="at">col =</span> col[<span class="fu">as.factor</span>(yy<span class="sc">$</span>samples<span class="sc">$</span>group)], <span class="at">outcex =</span> <span class="fl">0.3</span>,</span>
<span id="cb15-4"><a href="preprocessing-the-data.html#cb15-4" tabindex="-1"></a>    <span class="at">outcol =</span> col[<span class="fu">as.factor</span>(yy<span class="sc">$</span>samples<span class="sc">$</span>group)], <span class="at">labels =</span> <span class="fu">colnames</span>(yy.logcpm), <span class="at">main =</span> <span class="st">&quot;SgRNA abundance before filtering&quot;</span>,</span>
<span id="cb15-5"><a href="preprocessing-the-data.html#cb15-5" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">&quot;SgRNA representation (log2 cpm normalised reads)&quot;</span>)</span>
<span id="cb15-6"><a href="preprocessing-the-data.html#cb15-6" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topleft&quot;</span>, <span class="at">legend =</span> <span class="fu">levels</span>(<span class="fu">factor</span>(yy<span class="sc">$</span>samples<span class="sc">$</span>group)), <span class="at">col =</span> col, <span class="at">pch =</span> <span class="dv">19</span>,</span>
<span id="cb15-7"><a href="preprocessing-the-data.html#cb15-7" tabindex="-1"></a>    <span class="at">cex =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/library%20sizes%20before%20filtering-1.png" width="576" /></p>
</div>
<div id="removing-lowly-expressed-guides" class="section level2 unnumbered hasAnchor">
<h2>Removing lowly expressed guides<a href="preprocessing-the-data.html#removing-lowly-expressed-guides" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Guides lacking a substantial number of reads in any sample should be excluded from downstream analyses. There are multiple rationales for this choice. From a biological perspective, guides that fail to register at a biologically relevant level in any condition are deemed uninteresting and are consequently disregarded. From a statistical standpoint, the elimination of guides with low counts enhances the reliability of estimating the mean-variance relationship in the data. It also reduces the number of statistical tests required for subsequent analyses focused on differential abundance.</p>
<p>To implement this, here we used two filtering methods: one permissive and the other strict. For the rigorous approach, guides with low expression were filtered out, retaining as many guides as possible with meaningful counts. This was achieved using the filterByExpr() function in the edgeR package, a tool commonly applied in RNA-seq experiments for automated gene filtering. Initially, the count matrix contained 122,411 guides. After applying filterByExpr(), 113,395 guides (93%) were removed due to their low presence, leaving 9,016 guides for further analysis. It’s worth noting that filterByExpr() often retains only one guide per gene, which is suboptimal for CRISPR screens, where the collective behavior of multiple guides targeting the same gene is essential for validation.</p>
<p>To address this limitation, we used a more permissive filtering approach. This method involved defining guide-wise threshold (th1) and sample-wise threshold (th2). Th1 was computed using count per million (CPM) normalized values, while th2 was determined as the average number of samples in each condition. This approach aims to strike a balance, allowing for the inclusion of as many guides as possible targeting the same gene, facilitating a more robust assessment of the collective impact of guides in CRISPR screens. Consequently, with the permissive approach, we retained 58171 (48%) of the guides.</p>
<div id="remove-non-targeting-guides" class="section level3 hasAnchor" number="0.0.1">
<h3><span class="header-section-number">0.0.1</span> Remove Non-targeting guides<a href="preprocessing-the-data.html#remove-non-targeting-guides" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First we remove 1000 Non-targeting sgRNAs.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="preprocessing-the-data.html#cb16-1" tabindex="-1"></a>d.raw <span class="ot">&lt;-</span> d</span>
<span id="cb16-2"><a href="preprocessing-the-data.html#cb16-2" tabindex="-1"></a><span class="fu">dim</span>(d <span class="ot">&lt;-</span> d[<span class="sc">!</span>d<span class="sc">$</span>genes<span class="sc">$</span>Symbol <span class="sc">%in%</span> <span class="fu">grep</span>(<span class="st">&quot;Non&quot;</span>, d<span class="sc">$</span>genes<span class="sc">$</span>Symbol, <span class="at">value =</span> <span class="cn">TRUE</span>), ])</span></code></pre></div>
<pre><code>[1] 121411      6</code></pre>
</div>
<div id="strict-filtering" class="section level3 unnumbered hasAnchor">
<h3>Strict filtering<a href="preprocessing-the-data.html#strict-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="preprocessing-the-data.html#cb18-1" tabindex="-1"></a>keep.exprs <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(d, <span class="at">group =</span> d<span class="sc">$</span>samples<span class="sc">$</span>group)</span>
<span id="cb18-2"><a href="preprocessing-the-data.html#cb18-2" tabindex="-1"></a><span class="fu">table</span>(keep.exprs)</span></code></pre></div>
<pre><code>keep.exprs
 FALSE   TRUE 
112601   8810 </code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="preprocessing-the-data.html#cb20-1" tabindex="-1"></a><span class="fu">dim</span>(d.filtered.s <span class="ot">&lt;-</span> d[keep.exprs, , <span class="at">keep.lib.sizes =</span> <span class="cn">FALSE</span>])</span></code></pre></div>
<pre><code>[1] 8810    6</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="preprocessing-the-data.html#cb22-1" tabindex="-1"></a>genes_sgrna <span class="ot">&lt;-</span> d.filtered.s<span class="sc">$</span>genes<span class="sc">$</span>Symbol</span>
<span id="cb22-2"><a href="preprocessing-the-data.html#cb22-2" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(genes_sgrna))</span></code></pre></div>
<pre><code>[1] 7277</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="preprocessing-the-data.html#cb24-1" tabindex="-1"></a>nsgrna <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">table</span>(genes_sgrna))</span>
<span id="cb24-2"><a href="preprocessing-the-data.html#cb24-2" tabindex="-1"></a>nsgrna</span></code></pre></div>
<pre><code>
   1    2    3    4    5    6 
5912 1215  135   13    1    1 </code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="preprocessing-the-data.html#cb26-1" tabindex="-1"></a><span class="fu">barplot</span>(nsgrna, <span class="at">col =</span> <span class="st">&quot;#FF9999&quot;</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">7000</span>), <span class="at">border =</span> <span class="cn">NA</span>, <span class="at">main =</span> <span class="st">&quot;The distribution of number of sgRNAs per gene </span><span class="sc">\n</span><span class="st"> (After strict filtering)&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/strict%20filtering%20guide%20distribution-1.png" width="576" /></p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="preprocessing-the-data.html#cb27-1" tabindex="-1"></a>yy <span class="ot">&lt;-</span> d.filtered.s[, <span class="fu">order</span>(d.filtered.s<span class="sc">$</span>samples<span class="sc">$</span>group, <span class="fu">colSums</span>(d.filtered.s<span class="sc">$</span>counts))]</span>
<span id="cb27-2"><a href="preprocessing-the-data.html#cb27-2" tabindex="-1"></a>yy.logcpm <span class="ot">&lt;-</span> <span class="fu">cpm</span>(yy<span class="sc">$</span>counts, <span class="at">log =</span> T)</span>
<span id="cb27-3"><a href="preprocessing-the-data.html#cb27-3" tabindex="-1"></a><span class="fu">boxplot</span>(yy.logcpm, <span class="at">las =</span> <span class="dv">2</span>, <span class="at">pch =</span> <span class="dv">19</span>, <span class="at">cex.axis =</span> <span class="fl">0.5</span>, <span class="at">col =</span> col[<span class="fu">as.factor</span>(yy<span class="sc">$</span>samples<span class="sc">$</span>group)],</span>
<span id="cb27-4"><a href="preprocessing-the-data.html#cb27-4" tabindex="-1"></a>    <span class="at">outcex =</span> <span class="fl">0.3</span>, <span class="at">outcol =</span> col[<span class="fu">as.factor</span>(yy<span class="sc">$</span>samples<span class="sc">$</span>group)], <span class="at">labels =</span> <span class="fu">colnames</span>(yy.logcpm),</span>
<span id="cb27-5"><a href="preprocessing-the-data.html#cb27-5" tabindex="-1"></a>    <span class="at">main =</span> <span class="st">&quot;SgRNA abundance after strict filtering&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;SgRNA representation (log2 cpm normalised reads)&quot;</span>)</span>
<span id="cb27-6"><a href="preprocessing-the-data.html#cb27-6" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topleft&quot;</span>, <span class="at">legend =</span> <span class="fu">levels</span>(<span class="fu">factor</span>(yy<span class="sc">$</span>samples<span class="sc">$</span>group)), <span class="at">col =</span> col, <span class="at">pch =</span> <span class="dv">19</span>,</span>
<span id="cb27-7"><a href="preprocessing-the-data.html#cb27-7" tabindex="-1"></a>    <span class="at">cex =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/library%20sizes%20after%20strict%20filtering-1.png" width="576" /></p>
</div>
<div id="permissive-filtering" class="section level3 unnumbered hasAnchor">
<h3>Permissive filtering<a href="preprocessing-the-data.html#permissive-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="preprocessing-the-data.html#cb28-1" tabindex="-1"></a>th1 <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">cpm</span>(<span class="dv">1</span>, <span class="fu">mean</span>(d<span class="sc">$</span>samples<span class="sc">$</span>lib.size)))</span>
<span id="cb28-2"><a href="preprocessing-the-data.html#cb28-2" tabindex="-1"></a>th1  <span class="co"># average count per million reads (1e6/mean(d$samples$lib.size))</span></span></code></pre></div>
<pre><code>[1] 2.90183</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="preprocessing-the-data.html#cb30-1" tabindex="-1"></a>th2 <span class="ot">&lt;-</span> <span class="dv">2</span>  <span class="co"># average number of sample per condition</span></span>
<span id="cb30-2"><a href="preprocessing-the-data.html#cb30-2" tabindex="-1"></a>isexprs <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">cpm</span>(d) <span class="sc">&gt;</span> th1) <span class="sc">&gt;=</span> th2</span>
<span id="cb30-3"><a href="preprocessing-the-data.html#cb30-3" tabindex="-1"></a><span class="fu">table</span>(isexprs)</span></code></pre></div>
<pre><code>isexprs
FALSE  TRUE 
64038 57373 </code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="preprocessing-the-data.html#cb32-1" tabindex="-1"></a><span class="fu">dim</span>(d.filtered.p <span class="ot">&lt;-</span> d[isexprs, ])</span></code></pre></div>
<pre><code>[1] 57373     6</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="preprocessing-the-data.html#cb34-1" tabindex="-1"></a>genes_sgrna <span class="ot">&lt;-</span> d.filtered.p<span class="sc">$</span>genes<span class="sc">$</span>Symbol</span>
<span id="cb34-2"><a href="preprocessing-the-data.html#cb34-2" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(genes_sgrna))</span></code></pre></div>
<pre><code>[1] 19736</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="preprocessing-the-data.html#cb36-1" tabindex="-1"></a>nsgrna <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">table</span>(genes_sgrna))</span>
<span id="cb36-2"><a href="preprocessing-the-data.html#cb36-2" tabindex="-1"></a>nsgrna</span></code></pre></div>
<pre><code>
   1    2    3    4    5    6 
2672 4968 5933 4162 1688  313 </code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="preprocessing-the-data.html#cb38-1" tabindex="-1"></a><span class="fu">barplot</span>(nsgrna, <span class="at">col =</span> <span class="st">&quot;#FF9999&quot;</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">7000</span>), <span class="at">border =</span> <span class="cn">NA</span>, <span class="at">main =</span> <span class="st">&quot;The distribution of number of sgRNAs per gene </span><span class="sc">\n</span><span class="st"> (After permissive filtering)&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/permissive%20filtering%20guide%20distribution-1.png" width="576" /></p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="preprocessing-the-data.html#cb39-1" tabindex="-1"></a>yy <span class="ot">&lt;-</span> d.filtered.p[, <span class="fu">order</span>(d.filtered.p<span class="sc">$</span>samples<span class="sc">$</span>group, <span class="fu">colSums</span>(d.filtered.p<span class="sc">$</span>counts))]</span>
<span id="cb39-2"><a href="preprocessing-the-data.html#cb39-2" tabindex="-1"></a>yy.logcpm <span class="ot">&lt;-</span> <span class="fu">cpm</span>(yy<span class="sc">$</span>counts, <span class="at">log =</span> T)</span>
<span id="cb39-3"><a href="preprocessing-the-data.html#cb39-3" tabindex="-1"></a><span class="fu">boxplot</span>(yy.logcpm, <span class="at">las =</span> <span class="dv">2</span>, <span class="at">pch =</span> <span class="dv">19</span>, <span class="at">cex.axis =</span> <span class="fl">0.5</span>, <span class="at">col =</span> col[<span class="fu">as.factor</span>(yy<span class="sc">$</span>samples<span class="sc">$</span>group)],</span>
<span id="cb39-4"><a href="preprocessing-the-data.html#cb39-4" tabindex="-1"></a>    <span class="at">outcex =</span> <span class="fl">0.3</span>, <span class="at">outcol =</span> col[<span class="fu">as.factor</span>(yy<span class="sc">$</span>samples<span class="sc">$</span>group)], <span class="at">labels =</span> <span class="fu">colnames</span>(yy.logcpm),</span>
<span id="cb39-5"><a href="preprocessing-the-data.html#cb39-5" tabindex="-1"></a>    <span class="at">main =</span> <span class="st">&quot;SgRNA abundance after permissive filtering&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;SgRNA representation (log2 cpm normalised reads)&quot;</span>)</span>
<span id="cb39-6"><a href="preprocessing-the-data.html#cb39-6" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topleft&quot;</span>, <span class="at">legend =</span> <span class="fu">levels</span>(<span class="fu">factor</span>(yy<span class="sc">$</span>samples<span class="sc">$</span>group)), <span class="at">col =</span> col, <span class="at">pch =</span> <span class="dv">19</span>,</span>
<span id="cb39-7"><a href="preprocessing-the-data.html#cb39-7" tabindex="-1"></a>    <span class="at">cex =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/library%20sizes%20after%20permissive%20filtering-1.png" width="576" /></p>
</div>
</div>
<div id="normalising-the-sgrna-counts" class="section level2 unnumbered hasAnchor">
<h2>Normalising the sgRNA counts<a href="preprocessing-the-data.html#normalising-the-sgrna-counts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>During the sample preparation or sequencing process, external factors that are not of biological interest can affect the expression of individual samples.
For example, samples processed in the first batch of an experiment can have higher expression overall when compared to samples processed in a second batch.
It is assumed that all samples should have a similar range and distribution of expression values.
Normalisation is required to ensure that the expression distributions of each sample are similar across the entire experiment.
Here we applied the upperquartile method from calcNormFactors function in edgeR. In the upper-quartile normalization method of Bullard et al (2010), <strong>Bullard JH, Purdom E, Hansen KD, Dudoit S. (2010) Evaluation of statistical methods for normalization and differential expression in mRNA-Seq experiments. BMC Bioinformatics 11, 94.</strong>, the scale factors are calculated from the 75% quantile of the counts for each library, after removing genes that are zero in all libraries.
When working with DGEList-objects, the normalisation factors are automatically stored.
For this dataset, the effect of upperquartile-normalisation is mild, as evident in the magnitude of the scaling factors, which are all not too far from 1 as we expected them to be.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="preprocessing-the-data.html#cb40-1" tabindex="-1"></a>yy.norm <span class="ot">&lt;-</span> d.filtered.s</span>
<span id="cb40-2"><a href="preprocessing-the-data.html#cb40-2" tabindex="-1"></a>yy.norm<span class="sc">$</span>samples<span class="sc">$</span>norm.factors <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(yy.norm<span class="sc">$</span>counts, <span class="at">method =</span> <span class="st">&quot;TMM&quot;</span>)</span>
<span id="cb40-3"><a href="preprocessing-the-data.html#cb40-3" tabindex="-1"></a>yy.norm<span class="sc">$</span>samples[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]  <span class="co"># TMM norm factors of strictly filtered data</span></span></code></pre></div>
<pre><code>            group lib.size norm.factors
Control_1 Control    56390    2.0995847
Control_2 Control    57402    1.8983192
ToxA_1       ToxA   283958    1.8621100
ToxA_2       ToxA   242030    1.9418813
ToxB_1       ToxB   298310    0.2919254
ToxB_2       ToxB   300952    0.2376826</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="preprocessing-the-data.html#cb42-1" tabindex="-1"></a>yy.norm<span class="sc">$</span>samples<span class="sc">$</span>norm.factors <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(yy.norm<span class="sc">$</span>counts <span class="sc">+</span> <span class="dv">100</span>, <span class="at">method =</span> <span class="st">&quot;upperquartile&quot;</span>)</span>
<span id="cb42-2"><a href="preprocessing-the-data.html#cb42-2" tabindex="-1"></a>yy.norm<span class="sc">$</span>samples[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]  <span class="co"># Upperquartile norm factors of strict filtered data</span></span></code></pre></div>
<pre><code>            group lib.size norm.factors
Control_1 Control    56390    1.0911486
Control_2 Control    57402    1.0899719
ToxA_1       ToxA   283958    1.0935077
ToxA_2       ToxA   242030    1.0929344
ToxB_1       ToxB   298310    0.8436597
ToxB_2       ToxB   300952    0.8339069</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="preprocessing-the-data.html#cb44-1" tabindex="-1"></a>yy.norm <span class="ot">&lt;-</span> d.filtered.p</span>
<span id="cb44-2"><a href="preprocessing-the-data.html#cb44-2" tabindex="-1"></a>yy.norm<span class="sc">$</span>samples<span class="sc">$</span>norm.factors <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(yy.norm<span class="sc">$</span>counts, <span class="at">method =</span> <span class="st">&quot;TMM&quot;</span>)</span>
<span id="cb44-3"><a href="preprocessing-the-data.html#cb44-3" tabindex="-1"></a>yy.norm<span class="sc">$</span>samples[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]  <span class="co"># TMM norm factors of permissive filtered data</span></span></code></pre></div>
<pre><code>            group lib.size norm.factors
Control_1 Control   338214    1.3660602
Control_2 Control   337711    1.3232627
ToxA_1       ToxA   344188    1.2026100
ToxA_2       ToxA   335848    1.1862281
ToxB_1       ToxB   356569    0.6792538
ToxB_2       ToxB   355131    0.5708986</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="preprocessing-the-data.html#cb46-1" tabindex="-1"></a>yy.norm<span class="sc">$</span>samples<span class="sc">$</span>norm.factors <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(yy.norm<span class="sc">$</span>counts <span class="sc">+</span> <span class="dv">100</span>, <span class="at">method =</span> <span class="st">&quot;upperquartile&quot;</span>)</span>
<span id="cb46-2"><a href="preprocessing-the-data.html#cb46-2" tabindex="-1"></a>yy.norm<span class="sc">$</span>samples[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]  <span class="co"># Upperquartile norm factors of permissive filtered data</span></span></code></pre></div>
<pre><code>            group lib.size norm.factors
Control_1 Control   338214    1.0326478
Control_2 Control   337711    1.0332658
ToxA_1       ToxA   344188    0.9818947
ToxA_2       ToxA   335848    1.0031259
ToxB_1       ToxB   356569    0.9801216
ToxB_2       ToxB   355131    0.9708126</code></pre>
</div>
<div id="mds-plots" class="section level2 unnumbered hasAnchor">
<h2>MDS plots<a href="preprocessing-the-data.html#mds-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="preprocessing-the-data.html#cb48-1" tabindex="-1"></a>cols <span class="ot">&lt;-</span> col</span>
<span id="cb48-2"><a href="preprocessing-the-data.html#cb48-2" tabindex="-1"></a><span class="fu">plotMDS</span>(yy.norm, <span class="at">col =</span> cols[<span class="fu">factor</span>(yy.norm<span class="sc">$</span>samples<span class="sc">$</span>group)], <span class="at">main =</span> <span class="st">&quot;MDS plot before normalisation&quot;</span>,</span>
<span id="cb48-3"><a href="preprocessing-the-data.html#cb48-3" tabindex="-1"></a>    <span class="at">pch =</span> <span class="dv">16</span>, <span class="at">cex =</span> <span class="dv">2</span>)</span>
<span id="cb48-4"><a href="preprocessing-the-data.html#cb48-4" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;bottomright&quot;</span>, <span class="at">legend =</span> <span class="fu">levels</span>(<span class="fu">factor</span>(yy.norm<span class="sc">$</span>samples<span class="sc">$</span>group)), <span class="at">col =</span> cols,</span>
<span id="cb48-5"><a href="preprocessing-the-data.html#cb48-5" tabindex="-1"></a>    <span class="at">pch =</span> <span class="dv">16</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/mds%20plot%20normalisation-1.png" width="576" /></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="counting-the-sgrnas-with-rsubread.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-abundance-analysis-at-guide-level.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
