<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Setting up | Analysing CRISPR Screens with edgeR</title>
  <meta name="description" content="<p>This is a book version to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Setting up | Analysing CRISPR Screens with edgeR" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a book version to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Setting up | Analysing CRISPR Screens with edgeR" />
  
  <meta name="twitter:description" content="<p>This is a book version to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  

<meta name="author" content="GÃ¶knur Giner" />


<meta name="date" content="2023-10-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="counting-the-sgrnas-with-rsubread.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.30/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Analysing-CRISPR-screens-with-edgeR</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html"><i class="fa fa-check"></i>Setting up</a>
<ul>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#install-r"><i class="fa fa-check"></i>Install R</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#install-rstudio"><i class="fa fa-check"></i>Install RStudio</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#install-required-packages"><i class="fa fa-check"></i>Install required packages</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#load-required-packages"><i class="fa fa-check"></i>Load required packages</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#download-the-material"><i class="fa fa-check"></i>Download the material</a></li>
<li class="chapter" data-level="" data-path="setting-up.html"><a href="setting-up.html#create-necessary-directories"><i class="fa fa-check"></i>Create necessary directories</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html"><i class="fa fa-check"></i>Counting the sgRNAs with RSubread</a>
<ul>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html#build-an-index"><i class="fa fa-check"></i>Build an index</a></li>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html#align-the-fastq-files"><i class="fa fa-check"></i>Align the FASTQ files</a></li>
<li class="chapter" data-level="" data-path="counting-the-sgrnas-with-rsubread.html"><a href="counting-the-sgrnas-with-rsubread.html#create-a-dgelist-object"><i class="fa fa-check"></i>Create a DGEList object</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html"><i class="fa fa-check"></i>Preprocessing the data</a>
<ul>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#remove-control-guides"><i class="fa fa-check"></i>Remove control guides</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#check-guide-distribution"><i class="fa fa-check"></i>Check guide distribution</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#evaluate-the-uniformity"><i class="fa fa-check"></i>Evaluate the uniformity</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#remove-lowly-expressed-guides"><i class="fa fa-check"></i>Remove lowly expressed guides</a>
<ul>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#strict-filtering"><i class="fa fa-check"></i>Strict filtering</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#permissive-filtering"><i class="fa fa-check"></i>Permissive filtering</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#look-at-density-plots"><i class="fa fa-check"></i>Look at density plots</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#normalise-the-counts"><i class="fa fa-check"></i>Normalise the counts</a>
<ul>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#cpm-normalised-counts"><i class="fa fa-check"></i>CPM normalised counts</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#tmm-normalised-counts"><i class="fa fa-check"></i>TMM normalised counts</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i>Unsupervised clustering of samples</a></li>
<li class="chapter" data-level="" data-path="preprocessing-the-data.html"><a href="preprocessing-the-data.html#visualise-guide-abundance-profiles-of-samples"><i class="fa fa-check"></i>Visualise guide abundance profiles of samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html"><i class="fa fa-check"></i>Differential abundance analysis at guide level</a>
<ul>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#create-the-design-matrix"><i class="fa fa-check"></i>Create the design matrix</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#fit-a-model-when-there-is-no-biological-replicates"><i class="fa fa-check"></i>Fit a model when there is no biological replicates</a>
<ul>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#classic-edger"><i class="fa fa-check"></i>Classic edgeR</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#edger-glmlrt"><i class="fa fa-check"></i>edgeR glmLRT</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#fit-a-model-when-there-are-biological-replicates"><i class="fa fa-check"></i>Fit a model when there are biological replicates</a>
<ul>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#dispersion-estimates-and-biological-variation"><i class="fa fa-check"></i>Dispersion estimates and biological variation</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#edger-glmlrt-1"><i class="fa fa-check"></i>edgeR glmLRT</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-guide-level.html"><a href="differential-abundance-analysis-at-guide-level.html#edger-quasi"><i class="fa fa-check"></i>edgeR quasi</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="visualise-hit-guides.html"><a href="visualise-hit-guides.html"><i class="fa fa-check"></i>Visualise hit guides</a>
<ul>
<li class="chapter" data-level="" data-path="visualise-hit-guides.html"><a href="visualise-hit-guides.html#volcano-plot-toxa-vs-control"><i class="fa fa-check"></i>Volcano plot (ToxA vs Control)</a></li>
<li class="chapter" data-level="" data-path="visualise-hit-guides.html"><a href="visualise-hit-guides.html#volcano-plot-toxb-vs-control"><i class="fa fa-check"></i>Volcano plot (ToxB vs Control)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-gene-level.html"><a href="differential-abundance-analysis-at-gene-level.html"><i class="fa fa-check"></i>Differential abundance analysis at gene level</a>
<ul>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-gene-level.html"><a href="differential-abundance-analysis-at-gene-level.html#toxa-vs-control"><i class="fa fa-check"></i>ToxA vs Control</a></li>
<li class="chapter" data-level="" data-path="differential-abundance-analysis-at-gene-level.html"><a href="differential-abundance-analysis-at-gene-level.html#toxb-vs-control"><i class="fa fa-check"></i>ToxB vs Control</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysing CRISPR Screens with edgeR</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="setting-up" class="section level1 unnumbered hasAnchor">
<h1>Setting up<a href="setting-up.html#setting-up" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="install-r" class="section level2 unnumbered hasAnchor">
<h2>Install R<a href="setting-up.html#install-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>R can be installed from the <a href="http://www.r-project.org/">R-project website</a>.</p>
<p>Direct downloads for all platforms can be downloaded at the links below:</p>
<p><a href="https://cran.r-project.org/bin/macosx/">Mac</a></p>
<p><a href="https://cran.r-project.org/bin/windows/base/">Windows</a></p>
<p><a href="https://cran.r-project.org/bin/linux/ubuntu/fullREADME.html">Linux</a></p>
</div>
<div id="install-rstudio" class="section level2 unnumbered hasAnchor">
<h2>Install RStudio<a href="setting-up.html#install-rstudio" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>RStudio can be installed from the <a href="http://www.rstudio.com/">RStudio website</a>.</p>
<p>RStudio can be downloaded for all platforms at the link below:</p>
<p><a href="https://rstudio.com/products/rstudio/download/" class="uri">https://rstudio.com/products/rstudio/download/</a></p>
</div>
<div id="install-required-packages" class="section level2 unnumbered hasAnchor">
<h2>Install required packages<a href="setting-up.html#install-required-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="setting-up.html#cb1-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb1-2"><a href="setting-up.html#cb1-2" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;edgeR&quot;</span>)</span>
<span id="cb1-3"><a href="setting-up.html#cb1-3" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;GenomicAlignments&quot;</span>)</span>
<span id="cb1-4"><a href="setting-up.html#cb1-4" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;Rsubread&quot;</span>)</span>
<span id="cb1-5"><a href="setting-up.html#cb1-5" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;Biostrings&quot;</span>)</span>
<span id="cb1-6"><a href="setting-up.html#cb1-6" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;AnnotationDbi&quot;</span>)</span>
<span id="cb1-7"><a href="setting-up.html#cb1-7" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;org.Hs.eg.db&quot;</span>)</span>
<span id="cb1-8"><a href="setting-up.html#cb1-8" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb1-9"><a href="setting-up.html#cb1-9" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;DT&quot;</span>)</span>
<span id="cb1-10"><a href="setting-up.html#cb1-10" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggrepel&quot;</span>)</span>
<span id="cb1-11"><a href="setting-up.html#cb1-11" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;RColorBrewer&quot;</span>)</span>
<span id="cb1-12"><a href="setting-up.html#cb1-12" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;GO.db&quot;</span>)</span></code></pre></div>
</div>
<div id="load-required-packages" class="section level2 unnumbered hasAnchor">
<h2>Load required packages<a href="setting-up.html#load-required-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="setting-up.html#cb2-1" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb2-2"><a href="setting-up.html#cb2-2" tabindex="-1"></a><span class="fu">library</span>(GenomicAlignments)</span>
<span id="cb2-3"><a href="setting-up.html#cb2-3" tabindex="-1"></a><span class="fu">library</span>(Rsubread)</span>
<span id="cb2-4"><a href="setting-up.html#cb2-4" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb2-5"><a href="setting-up.html#cb2-5" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb2-6"><a href="setting-up.html#cb2-6" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb2-7"><a href="setting-up.html#cb2-7" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2-8"><a href="setting-up.html#cb2-8" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb2-9"><a href="setting-up.html#cb2-9" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb2-10"><a href="setting-up.html#cb2-10" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb2-11"><a href="setting-up.html#cb2-11" tabindex="-1"></a><span class="fu">library</span>(GO.db)</span></code></pre></div>
</div>
<div id="download-the-material" class="section level2 unnumbered hasAnchor">
<h2>Download the material<a href="setting-up.html#download-the-material" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The sequencing files and human GeCKO sgRNA library can be downloaded from the github repository by clicking on the following links.</p>
<ol style="list-style-type: decimal">
<li><p><a href="https://github.com/goknurginer/Analysing-CRISPR-screens-with-edgeR/tree/main/fasqfiles">FASTQ files</a></p></li>
<li><p><a href="https://github.com/goknurginer/Analysing-CRISPR-screens-with-edgeR/tree/main/sgRNA_library">GeCKO human library</a></p></li>
</ol>
</div>
<div id="create-necessary-directories" class="section level2 unnumbered hasAnchor">
<h2>Create necessary directories<a href="setting-up.html#create-necessary-directories" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="setting-up.html#cb3-1" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;./index&quot;</span>)</span>
<span id="cb3-2"><a href="setting-up.html#cb3-2" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;./RData&quot;</span>)</span>
<span id="cb3-3"><a href="setting-up.html#cb3-3" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;./tables&quot;</span>)</span>
<span id="cb3-4"><a href="setting-up.html#cb3-4" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;./figures&quot;</span>)</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="counting-the-sgrnas-with-rsubread.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/goknurginer/Analysing-CRISPR-screens-with-edgeRindex.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Analysing-CRISPR-screens-with-edgeR.pdf", "Analysing-CRISPR-screens-with-edgeR.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
